### DISCUSSION TITLE
Using async action creator without the thunk?

### DISCUSSION THREAD

**Discussion by hellos3b** [reduxjs/redux-toolkit#3302] [Q&A] [answered]
I know this is *redux* toolkit, but I maintain a lot of my state locally with `React.useReducer`, and I find the `createReducer` + `createAction` handy for that as well.

Similarly, I was wondering if there was a way to manually use the underlying action creators from `createAsyncThunk`, but without the thunk part of it? Since I don't have an actual store with middlewares I can't use the thunk, but the auto creation of the states would still be very handy

```ts
const Login = createAsyncAction("Login");

dispatch(Login.pending());
```

**Comment by phryneas** [comment, marked as answer]
Yes and no - the action creators expect a lot of extra "meta information" that you manually barely can give them.

Can I ask why you use `createReducer` + `createAction` and not just `createSlice`? That would also make that generation of the `pending` action etc. less work.

**Reply by hellos3b** [reply]
> Can I ask why you use createReducer + createAction and not just createSlice?

Hadn't even considered using a slice, you're right that it does alleviate my concerns, thanks
